SharedCode/Repository/DB/PokemonRepositoryLocal.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using SharedCode.Database;
+using SharedCode.Model;
+using SharedCode.Model.Api;
+using SharedCode.Model.DB;
+using SharedCode.Repository.Interfaces;
+using SharedCode.Util;
+using Unity;
+
+namespace SharedCode.Repository.DB
+{
+	public class PokemonRepositoryLocal : IPokemonRepositoryLocal
+	{
+		private readonly INetworkHandler NetworkHandler;
+
+        public IPokemonRepository Repository;
+
+        [Dependency]
+		public IDatabaseManager DatabaseManager;
+
+        public PokemonRepositoryLocal(IDatabaseManager databaseManager, IPokemonRepository repository)
+		{
+            this.Repository = repository;
+			this.DatabaseManager = databaseManager;
+            //BuildPokemonLocalList();
+		}
+
+		public async Task BuildPokemonLocalList()
+		{
+			if (DatabaseManager.checkTableExists("Pokemon"))
```

Comment:
As a suggestion you can add this to an enum and call it instead of writing the name every time